client/homebrew/brewRenderer/toolBar/toolBar.jsx

```diff
@@ -0,0 +1,124 @@
+require('./toolBar.less');
+const React = require('react');
+const { useState, useEffect } = React;
+const _ = require('lodash')
+
+const MAX_ZOOM = 300;
+const MIN_ZOOM = 10;
+
+const ToolBar = ({ onZoomChange, currentPage, onPageChange, totalPages })=>{
+
+	const [zoomLevel, setZoomLevel] = useState(100);
+	const [pageInput, setPageInput] = useState(currentPage);
+
+	useEffect(()=>{
+		onZoomChange(zoomLevel);
+	}, [zoomLevel]);
+
+	useEffect(()=>{
+		setPageInput(currentPage);
+	}, [currentPage])
+
+	const handleZoomChange = (delta)=>{
+		const zoomChange = _.clamp(zoomLevel + delta, MIN_ZOOM, MAX_ZOOM);
+
+		setZoomLevel(zoomChange);
+	};
+
+	const handlePageChange = (page)=>{
+		setPageInput(page);
+	};
+
+	const scrollToPage = (pageNumber) => {
+		pageNumber = _.clamp(pageNumber - 1, 0, totalPages - 1);
+		const iframe = document.getElementById('BrewRenderer');
+		if(iframe && iframe.contentWindow) {
+			const brewRenderer = iframe.contentWindow.document.querySelector('.brewRenderer');
+			if(brewRenderer) {
+				const pages = brewRenderer.querySelectorAll('.page');
+				pages[pageNumber]?.scrollIntoView({ block: 'start' });
+			}
+		}
+	};
+
+	return (
+		<div className='toolBar'>
+			<div className='group'>
+				<button
+					id='zoom-out'
+					className='tool'
+					onClick={()=>handleZoomChange(-20)}
+					disabled={zoomLevel <= MIN_ZOOM}
+				>
+					<i className='fas fa-magnifying-glass-minus' />
+				</button>
+				<input
+					id='zoom-slider'
+					className='range-input tool'
+					type='range'
+					name='zoom'
+					list='zoomLevels'
+					min={MIN_ZOOM}
+					max={MAX_ZOOM}
+					step='1'
+					value={zoomLevel}
+					onChange={(e)=>{setZoomLevel(parseInt(e.target.value));}}
+				/>
+				<datalist id='zoomLevels'>
+					<option value='100' />
+				</datalist>
+
+				<button
+					id='zoom-in'
+					className='tool'
+					onClick={()=>handleZoomChange(20)}
+					disabled={zoomLevel >= MAX_ZOOM}
+				>
+					<i className='fas fa-magnifying-glass-plus' />
+				</button>
+			</div>
+
+			<div className='group'>
+				<button
+					id='previous-page'
+					className='previousPage tool'
+					onClick={()=>scrollToPage(pageInput - 1)}
+					disabled={pageInput <= 1}
+				>
+					<i className='fas fa-arrow-left'></i>
+				</button>
+
+				<div className='tool'>
+					<input
+						id='page-input'
+						class='text-input'
+						type='text'
+						name='page'
+						inputMode='numeric'
+						pattern='[0-9]'
+						value={pageInput}
+						onChange={(e)=>{
+							handlePageChange(e.target.value == false ? e.target.value : parseInt(e.target.value));}}
```

Comment:
For testing purposes, in what case would e.target.value == false, and what is the expected behavior of setting `pageInput = false`?

Reply:
If the value entered is zero, it would return false, right?  And so if it's false, it will populate the field with it's value, which is the pageNum state which remains what it was (since it hasn't been updated yet).

Reply:
Its saving the input value whether its 0 or not. The only difference is we use parseInt for numbers > 0 but not for 0?

Reply:
okay yeah, I'm not really sure what i was defending.